service/src/main/java/org/databiosphere/workspacedataservice/InstanceInitializerBean.java

```
@@ -288,14 +298,16 @@ private void restoreFromRemoteBackup(String backupFileName, UUID cloneJobId) {
   */
   private void initializeDefaultInstance() {
     try {
+      // Don't create schema if in the middle of cloning
+      boolean lockAquired = lock.tryLock();
```

I don't think we need a lock here. The only case in which `initializeDefaultInstance` is called in a WDS where `SOURCE_WORKSPACE_ID` is populated is if the WDS attempted to restore a clone, but failed.

---

This is not the case with replicas. If one replica is in the middle of cloning, the second replica that starts after that but before the first is finished will have isInCloneMode return false and call initializeDefaultInstance. Even if I move the cloneDao.cloneExistsForWorkspace check to initCloneMode, we'd return false out of there and immediately call initializeDefaultInstance.

---

I know it's not the case currently - I really think we need to change that behavior, though. With `cloneDao.cloneExistsForWorkspace` moving out of `isInCloneMode`, it'll change what `isInCloneMode` returns.

https://lucid.app/lucidchart/fd59bf3a-da3b-4c35-bf5e-cac58c498cb2/edit?invitationId=inv_192a4441-f000-49b0-a97c-2d07889ba58e&page=0_0# is still how I think the overall flow should work!

---

the TL;DR of how I think it should work:
| If | Then |
|--------|--------|
| `SOURCE_WORKSPACE_ID` undefined | create default instance; swallow errors if that instance already exists |
| `SOURCE_WORKSPACE_ID` defined; can't acquire lock | Do nothing |
| `SOURCE_WORKSPACE_ID` defined; lock acquired; row exists in clone table | Do nothing |
| `SOURCE_WORKSPACE_ID` defined; lock acquired; no row in clone table | Perform clone stuff |

---

Noting here: This will be refactored in follow up work in this ticket https://broadworkbench.atlassian.net/browse/AJ-1363